routes/api/v1/signup.js

```diff
@@ -0,0 +1,30 @@
+var express = require('express');
+var router = express.Router();
+
+router.post('/', (req, res, next) => {
+  var db = req.app.get('db')
+  var user = req.body
+
+  if(user.username && user.password) {
+    db('users').where({username: user.username})
+    .then( (existingUser) => {
+      if(existingUser[0]) {
+        return res.json({errorMessage: 'Username already exists'})
+      } else {
+        //hash password with bcrypt before saving
+        db('users').insert(user)
+        .then( () => {
+          //generate JWT
+          //return JWT
+          return res.json()
+        });
+      }
+    });
+    }else{
+      var err = new Error('Not found');
+      err.status = 400;
+      next(err);
+    }
```

Comment:
@wilmoore 
Now if its logic dealing with models (CRUD) I was thinking of a models folder like models/user.js and inside there have my CRUD functionality. But if it is logic that can be shared with all models you are saying to put that in a lib folder? Or are you saying inside lib put my models folder, similar to Rails app folder? sorry for the confusion. I have been looking for examples of how node folders are structured. Seems like everyone has their own opinion on the matter. Thanks for your input!